bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/Bookie.java

```diff
@@ -1141,6 +1145,40 @@ private void addEntryInternal(LedgerDescriptor handle, ByteBuf entry,
         getJournal(ledgerId).logAddEntry(entry, ackBeforeSync, cb, ctx);
     }
 
+    /**
+     * Force write on the journal assigned to the given ledger.
+     * It works like a regular addEntry with ackBeforeSync=false but without really
+     * writing to disk.
+     */
+    private void forceLedgerInternal(LedgerDescriptor handle,
+                                     WriteCallback cb, Object ctx, byte[] masterKey)
+            throws IOException, BookieException {
+        long ledgerId = handle.getLedgerId();
+
+
+        if (masterKeyCache.get(ledgerId) == null) {
```

Comment:
Can we avoid code duplication here? by introducing another method with a sync flag?

Reply:
I did not introduce the nee method in order not to change addEntry and focus on this change.
I will do if you prefer. No problem.
I thought it would be a follow refactor/enhancement

Reply:
you don't need to change addEntry, but you can abstract line 1156 to line 1173 as a common method? so this common method can be called in addEntry and forceLedgerInternal, rather than duplicating code.

Reply:
Wil do